backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/types/predicate/impl/ElasticsearchFieldPredicateBuilderFactory.java

```diff
@@ -29,6 +30,9 @@
 	 */
 	boolean isDslCompatibleWith(ElasticsearchFieldPredicateBuilderFactory other);
 
+	MatchIdPredicateBuilder<ElasticsearchSearchPredicateBuilder> createMatchIdPredicateBuilder(
```

Comment:
This is unnecessary; you can remove this method and its implementations.

To be clear:

* `SearchPredicateBuilderFactory` is the entry point for the engine to retrieve predicate builders. It's implemented by the backend.
* In some cases, for field-scoped predicates (match, range), `SearchPredicateBuilderFactory` implementations delegate to a field-scoped, backend-specific `ElasticsearchFieldPredicateBuilderFactory` or `LuceneFieldPredicateBuilderFactory`. But that's only when the predicate can have a different behavior depending on the field being targeted: the `match` predicate will accept different input based on whether you target an integer or `LocalDate` field, for example, but this concept doesn't make sense for the `bool` predicate, which doesn't really "target" a field.

So in the case of your ID predicate, you can see that you don't need to change the `ElasticsearchFieldPredicateBuilderFactory`, only the `SearchPredicateBuilderFactory` (which you already did).

I'll try to add some javadoc in a separate PR to make that clearer.